tests/serverpod_test_server/test_integration/inheritance_test.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:serverpod_test_server/test_util/test_serverpod.dart';
+import 'package:test/test.dart';
+
+Future<void> _deleteAll(Session session) async {
+  await ParentClass.db
+      .deleteWhere(session, where: (element) => Constant.bool(true));
+}
+
+void main() async {
+  var session = await IntegrationTestServer().session();
+
+  tearDownAll(() async => await _deleteAll(session));
+
+  test(
+      'Given a class extends another class, then the child-class is a sub-type parent-class',
+      () {
+    var childClass = ChildClass(
+      grandParentField: 'grandParentField',
+      parentField: 'parentField',
+      childField: 2,
+    );
+
+    expect(childClass, isA<ParentClass>());
+  });
+
+  test(
+      'Given a ChildClass is instantiated and inserted into the ParentClass table, then the ChildClass should be retrievable from the ParentClass table',
```

Comment:
Suggestion: Use the same phrasing for all tests.

Here you use ChildClass, in the test above you use child-class. Please use one type.

Reply:
Suggestion: Use Given, When, Then structure.
```suggestion
      'Given an instantiated ChildClass when inserted into the ParentClass table, then the ChildClass should be retrievable from the ParentClass table',
```

The `When` part is phrased differently here.